crates/snapbox/src/assert.rs

```diff
@@ -181,7 +181,7 @@ impl Assert {
         // On `expected` being an error, make a best guess
         if expected
             .as_ref()
-            .map(|d| d.as_str().is_some())
+            .map(|d| d.try_string().is_ok())
             .unwrap_or(true)
```

Comment:
Since this is unrelated to what `try_string` is going to be used for, let's revert